From 2862d2925a5ec2b91b05dd0ff2ea3ff1ba05cec1 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Tue, 24 Jul 2018 11:41:38 +0200
Subject: [PATCH 4/4] Updater: Use PowerManager instead of Intent.ACTION_REBOOT
 broadcast

* We are getting permission denial when sending Intent.ACTION_REBOOT
  broadcast, using pm.reboot() instead works just fine.

Change-Id: I38d0ad8cc5f4c9362fd5baf7476e14bff56616dd
---
 src/org/lineageos/updater/UpdatesListAdapter.java | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/org/lineageos/updater/UpdatesListAdapter.java b/src/org/lineageos/updater/UpdatesListAdapter.java
index 4fb483f..3c71203 100644
--- a/src/org/lineageos/updater/UpdatesListAdapter.java
+++ b/src/org/lineageos/updater/UpdatesListAdapter.java
@@ -15,11 +15,13 @@
  */
 package org.lineageos.updater;
 
+import android.content.Context;
 import android.content.Intent;
 import android.content.IntentFilter;
 import android.content.SharedPreferences;
 import android.content.res.Resources;
 import android.os.BatteryManager;
+import android.os.PowerManager;
 import android.preference.PreferenceManager;
 import android.support.design.widget.Snackbar;
 import android.support.v7.app.AlertDialog;
@@ -382,8 +384,11 @@ public class UpdatesListAdapter extends RecyclerView.Adapter<UpdatesListAdapter.
             case REBOOT: {
                 button.setText(R.string.reboot);
                 button.setEnabled(enabled);
-                clickListener = enabled ?
-                        view -> mActivity.sendBroadcast(new Intent(Intent.ACTION_REBOOT)) : null;
+                clickListener = enabled ? view -> {
+                    PowerManager pm =
+                            (PowerManager) mActivity.getSystemService(Context.POWER_SERVICE);
+                    pm.reboot(null);
+                } : null;
             }
             break;
             default:
-- 
2.17.1

